fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/VehicleFlowGenerator.java

```diff
@@ -418,29 +418,29 @@ private void createVehicle(SpawningFramework framework, String name, String grou
             return;
         }
         // if no group is defined in vehicle definition take group declared in prototype
-        group = ObjectUtils.defaultIfNull(group, type.getGroup());
+        String spawningGroup = ObjectUtils.defaultIfNull(group, type.getGroup());
 
         VehicleDeparture vehicleDeparture = new VehicleDeparture.Builder(route)
                 .departureLane(laneSelectionMode, lane, pos)
-                .departureConnection(departureConnectionIndex)
+                .departureConnection(departConnectionIndex)
                 .departureSpeed(departureSpeedMode, departSpeed)
                 .create();
 
         Interaction interaction;
         if (origin != null) {
             interaction = new RoutelessVehicleRegistration(
-                    framework.getTime(), name, group, type.getAppList(), vehicleDeparture, type.convertType(), odInfo
+                    framework.getTime(), name, spawningGroup, type.getAppList(), vehicleDeparture, type.convertType(), odInfo
             );
         } else {
-            interaction = new VehicleRegistration(framework.getTime(), name, group, type.getAppList(), vehicleDeparture,
+            interaction = new VehicleRegistration(framework.getTime(), name, spawningGroup, type.getAppList(), vehicleDeparture,
                     type.convertTypeAndVaryParameters(randomNumberGenerator)
             );
         }
 
         try {
-            LOG.info("Creating Vehicle. time={}, name={}, route={}, laneSelectionMode={}, lane={}, departureConnectionIndex{}, pos={}, "
-                            + "type={}, departSpeed={}, apps={}",
-                    framework.getTime(), name, route, laneSelectionMode, lane, departureConnectionIndex, pos, type, departSpeed, type.getAppList());
+            LOG.info("Creating Vehicle. time={}, name={}, route={}, laneSelectionMode={}, lane={}, departConnectionIndex={}, pos={}, type={}, departSpeed={}, apps={}",
+                    framework.getTime(), name, route, laneSelectionMode, lane,
+                    departConnectionIndex, pos, type.getPrototype(), departSpeed, type.getAppList());
```

Comment:
Sure this is not part of your change, but now as we stumbled upon it.. I dont like the following method names:

*  `getPrototype()` - Given the name I would expect a prototype object, but it returns just a string. Maybe change to `getPrototypeName()`?
* `getAppList()` - meh, having "list" in the method name is ugly, maybe `getApplications()` is better?

Reply:
I also don't like how the field `applications` in `UnitSpawner` is handled. It can be null but if it is null a new list instance is created. I rather would prefer to instantiate a list by default and fill it with all non-null values. When returning the list in `getAppList`, it could just return `Collections.unmodifiableList(applications)`.

Reply:
> I also don't like how the field `applications` in `UnitSpawner` is handled. It can be null but if it is null a new list instance is created. I rather would prefer to instantiate a list by default and fill it with all non-null values. When returning the list in `getAppList`, it could just return `Collections.unmodifiableList(applications)`.

While implementing this I realized that this change will break the existing functionality a little bit:
* prior the `fillInPrototype` method in `UnitSpawner` was checking if the application list was `null` prior to filling in the prototype-apps
* with the suggested change we will have to check for `applications.isEmpty()` as we'll always have an empty list for `applications`
* so prior to this it was possible to omit all prototype apps by defining an empty applications-list, with the change this is no longer possible

I still prefer the change though. Defining a vehicle of a specific type without inheriting it's apps now requires either a) a dummy app or b) a second definition of the prototype without apps

Reply:
Whoops, I didn't see that either. 

> Defining a vehicle of a specific type without inheriting it's apps now requires either a) a dummy app or b) a second definition of the prototype without apps.

I could live with the proposed change, as I cannot remember that we ever used a prototype but replaced the inherited applications with an empty list. I did not even know this was possible. So please do that and we'll see if something brakes 🤞.